src/app/clusters/closure-dimension-server/closure-dimension-cluster-logic.cpp

```diff
@@ -184,6 +399,25 @@ CHIP_ERROR ClusterLogic::SetUnitRange(const DataModel::Nullable<Structs::UnitRan
     // Return error if unitRange is invalid
     VerifyOrReturnError(unitRange.Value().min <= unitRange.Value().max, CHIP_ERROR_INVALID_ARGUMENT);
 
+    ClosureUnitEnum unit;
+    ReturnErrorOnFailure(GetUnit(unit));
+
+    // If Unit is Millimeter , Range values SHALL contain unsigned values from 0 to 32767 only
+    if (unit == ClosureUnitEnum::kMillimeter)
+    {
+        VerifyOrReturnError(unitRange.Value().min >= 0 && unitRange.Value().min <= 32767, CHIP_ERROR_INVALID_ARGUMENT);
+        VerifyOrReturnError(unitRange.Value().max >= 0 && unitRange.Value().max <= 32767, CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
This doesnt validated that min <= max.
```suggestion
        VerifyOrReturnError(unitRange.Value().min >= 0 && unitRange.Value().min <= 32767, CHIP_ERROR_INVALID_ARGUMENT);
        VerifyOrReturnError(unitRange.Value().min  <= unitRange.Value().max  && unitRange.Value().max <= 32767, CHIP_ERROR_INVALID_ARGUMENT);
```

Reply:
this check is done at line number 400

Reply:
Can you provide the link / snipit? i dont see it...